lib/configuration/connection_configuration.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const toml = require('toml');
+const os = require('os');
+const fsPromises = require('fs/promises');
+const { validateOnlyUserReadWritePermission } = require('../file_transfer_agent/file_util');
+
+function defaultIfNotSet(value, defaultValue) {
+  if (value === null || typeof value === 'undefined' || value === '') {
+    return defaultValue;
+  } else {
+    return value;
+  }
+}
+
+async function laodConnectionConfiguration() {
+  const path = defaultIfNotSet(process.env.SNOWFLAKE_HOME, os.homedir() + '/.snowflake/');
+  const filePath = path + 'connections.toml';
+  validateOnlyUserReadWritePermission(filePath);
```

Comment:
it's async function - we should await

please check also other method calls

Reply:
Fixed